numba/tests/test_array_manipulation.py

```diff
@@ -740,6 +746,78 @@ def check(x):
         self.assertIn("The argument to np.shape must be array-like",
                       str(raises.exception))
 
+    def test_flatnonzero_basic(self):
+        # these tests should pass in all numpy versions
+        pyfunc = numpy_flatnonzero
+        cfunc = jit(nopython=True)(pyfunc)
+
+        def a_variations():
+            yield np.arange(-5, 5)
+            yield np.full(5, fill_value=0)
+            yield np.array([])
+            a = self.random.randn(100)
+            a[np.abs(a) > 0.2] = 0.0
+            yield a
+            yield a.reshape(5, 5, 4)
+            yield a.reshape(50, 2, order='F')
+            yield a.reshape(25, 4)[1::2]
+            yield a * 1j
+
+        for a in a_variations():
+            expected = pyfunc(a)
+            got = cfunc(a)
+            self.assertPreciseEqual(expected, got)
+
+    @staticmethod
+    def array_like_variations():
+        yield ((1.1, 2.2), (3.3, 4.4), (5.5, 6.6))
+        yield (0.0, 1.0, 0.0, -6.0)
+        yield ([0, 1], [2, 3])
+        yield ()
+        yield np.nan
+        yield 0
+        yield 1
+        yield False
+        yield (True, False, True)
+        yield 2 + 1j
+        # the following are not array-like, but numpy 1.15+ does not raise
+        yield None
+        # string support isn't present in Numba for Python 2.7
+        if sys.version_info >= (3,):
```

Comment:
Think 3.4 was minimum for unicode support so this condition probably ought to read `if not sys.version_info < (3,)`